src/System.Runtime.Numerics/tests/BigInteger/ToByteArray.cs

```diff
@@ -0,0 +1,181 @@
+        {
+            BigInteger bi;
+
+            bi = new BigInteger(-0x100112233L);
+            Assert.Equal(new byte[] { 0xcd, 0xdd, 0xee, 0xff, 0xfe }, bi.ToByteArray());
+
+            bi = new BigInteger(-0x8000112233L);
+            Assert.Equal(new byte[] { 0xcd, 0xdd, 0xee, 0xff, 0x7f, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0x3cd00112233L);
+            Assert.Equal(new byte[] { 0xcd, 0xdd, 0xee, 0xff, 0x32, 0xfc }, bi.ToByteArray());
+
+            bi = new BigInteger(-0xf92100112233L);
+            Assert.Equal(new byte[] { 0xcd, 0xdd, 0xee, 0xff, 0xde, 0x06, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0x749aa00112233L);
+            Assert.Equal(new byte[] { 0xcd, 0xdd, 0xee, 0xff, 0x55, 0xb6, 0xf8 }, bi.ToByteArray());
+
+            bi = new BigInteger(-0x80112200112233L);
+            Assert.Equal(new byte[] { 0xcd, 0xdd, 0xee, 0xff, 0xdd, 0xee, 0x7f, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0x7654321000112233L);
+            Assert.Equal(new byte[] { 0xcd, 0xdd, 0xee, 0xff, 0xef, 0xcd, 0xab, 0x89 }, bi.ToByteArray());
+
+            bi = new BigInteger(long.MinValue);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x80 }, bi.ToByteArray());
+
+            bi = new BigInteger(-0x100000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0x300000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0xfd }, bi.ToByteArray());
+
+            bi = new BigInteger(-0x8000000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x80 }, bi.ToByteArray());
+
+            bi = new BigInteger(-0xfe00000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x02, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0xff00000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x01, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0xfeff00000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x01, 0x01, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0xffff00000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0xfeffff00000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x01, 0xff }, bi.ToByteArray());
+
+            bi = new BigInteger(-0xffffff00000000L);
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0xff }, bi.ToByteArray());
+
+            bi = BigInteger.Parse("-18374686475376656384");
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x01, 0xff }, bi.ToByteArray());
+
+            bi = BigInteger.Parse("-18446744069414584320");
+            Assert.Equal(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0xff }, bi.ToByteArray());
```

Comment:
I realize this value doesn't fit in an Int64, but could you add a couple of Parse-based tests for much bigger values as well?

Also, we should probably verify not only the expected by array but that roundtripping works correctly.  Do we have any such tests elsewhere? e.g. passing bi back to the ctor and verifying that it and the original BigInteger are the same value?

Given that doing so would expand the number of lines per test case a bit, you might also consider doing this as a [Theory] with [InlineData] instead, such that we end up with a reusable test body applied to a bunch of one-line [InlineData] attributes that contain the input and the expected output.